### PR TITLE
Issue #1562: submitted email missing info

### LINKED ISSUE #1465: Display Federal Agency, if applicable, on the Domain Overview/Review/Manage pages
### Issue description

As a domain registrant/manager of a Federal domain
I want to see the Federal agency associated with my organization
So that I can confirm that this information is correct.

1) On the registrar domain overview page:
If the `organization type` is Federal,
Then display the Federal agency immediately above the Organization name.

2) On the domain request "review" and "manage" pages:
If the `organization type` is Federal,
Then display the Federal agency immediately above the Organization name.

3) On the submitted request email (in the SUMMARY OF YOUR DOMAIN REQUEST) section

### Acceptance criteria

- [ ] When viewing the Domain Overview page, the Federal agency is displayed immediately above the Organization name in the "Organization name and mailing address" section.
- [ ] When viewing a domain request's "review" or "manage" page, the Federal agency is displayed immediately above the Organization name in the "Organization name and mailing address" section.
- [ ] On the submitted request email (in the SUMMARY OF YOUR DOMAIN REQUEST section). Add federal agency under "Organization name and mailing address:" section

### Additional context

During the request process, if the requestor specifies that the domain is for a Federal org type, then they are prompted, by dropdown on the `/request/organization_contact` page, to select a Federal agency along with the org name and address information.

Similarly, when managing the domain, the Federal agency name is editable, by dropdown, on the `org-name-address` page. However, the Federal agency is not shown on the Domain Overview page as part of the Org contact. It makes sense that this conditional data be displayed there.

Source: [User feedback](https://docs.google.com/document/d/1M5foXX34qPc7R_J1uhBACHWUhg8WHwX3bB6nurvNNWE/edit#bookmark=id.s5dr58755312). 

### Links to other issues

_No response_

### LINKED ISSUE #1562: Email for "submitted" request: Info missing from request summary
### Current Behavior

Within the "submitted" domain request email, we show a "Summary of your domain request." However, that summary is missing some information (E.g. type of work, current website, etc). 

### Expected Behavior

The "submitted" email should list all submitted information for the request. It should match the "Review" page in the registrar.

**One exception:** Do not display "Requirements for operating .gov domains" field. This is a required checkbox to complete the request process. No need to call it out in the email.

### Steps to Reproduce

1. Trigger an auto-generated email for a "submitted" domain request
2. Review the "summary of your domain request" within the email and compare to the submitted request.

### Additional Context

Some fields only appear for certain Org Types. Ensure you test this for each Org Type.

NOTE: we're removing the "Summary of your domain request" from all  domain-request emails EXCEPT for the "submitted" email. That's why the "submitted" email is the only one mentioned in this ticket.

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #1562 
Resolves #1465 

## Changes

- Added conditional sections to email based on Type of organization
- Added No other employees from your organization section
- Added Federal agency to domain request review, domain request manage, and domain request submission email

## Context for reviewers

Only way to test is to create new domain requests with differing conditions (org types, etc) and submit the requests (with your email address as the email in Your contact information) and review emails.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Comment by michelle-rago** [comment]
@dave-kennedy-ecs The domain review page didn't show the "Current website" I entered (first screenshot). However, after I submitted the request, I do see the current website when I click the domain from the dashboard (second screenshot).

image

File: src/registrar/templates/application_review.html [hunk 1]
```diff
@@ -20,108 +20,158 @@ <h1> Review and submit your domain request </h1>
...
-          {% if step == Step.ORGANIZATION_ELECTION %}
-            {{ application.is_election_board|yesno:"Yes,No,Incomplete" }}
-          {% endif %}
-          {% if step == Step.ORGANIZATION_CONTACT %}
-            {% if application.organization_name %}
-              {% include "includes/organization_address.html" with organization=application %}
-            {% else %}
-              Incomplete
-            {% endif %}
+  <section class="summary-item margin-top-3">
```
Comment (zandercymatics) [inline src/registrar/templates/application_review.html:23]: Nice refactor

File: src/registrar/templates/application_review.html [hunk 2]
```diff
@@ -20,108 +20,158 @@ <h1> Review and submit your domain request </h1>
...
+            {% include "includes/summary_item.html" with title=title value=value heading_level=heading_level editable=True edit_link=application_url %}
+          {% endwith %}
+
+          {% if application.alternative_domains.all %}
+            <h3 class="margin-bottom-0 text-base-darker">Alternative domains</h3>
+            <ul class="usa-list margin-top-0">
               {% for site in application.alternative_domains.all %}
                 <li>{{ site.website }}</li>
               {% endfor %}
             </ul>
```
Comment (zandercymatics) [inline src/registrar/templates/application_review.html:120]: (Comment) This seems similar in principle to the other summary items, but with a header and slightly different classes. Its not something you should address in this PR (Too much effort for too little reward) but I think summary_item could be genericized enough to handle this edge case

**Comment by dave-kennedy-ecs** [comment]
> @dave-kennedy-ecs The domain review page didn't show the "Current website" I entered (first screenshot). However, after I submitted the request, I do see the current website when I click the domain from the dashboard (second screenshot). The current website did appear in the email confirmation.
> 
> Let me know when it's okay for me to withdraw this domain request so I can submit another one with a different org type.
> 
> image
> 
> **Second screenshot**
> 
> image

@michelle-rago the current websites has been fixed, can you test again?

**Comment by michelle-rago** [comment]
There's an issue with "Other employees..." 

For "Other employees..." I said "no" and entered some text. 

This is the pre-submission review page with what I entered in that field:
image

The post-submission review page (https://getgov-dk.app.cloud.gov/application/220) just says "None."
image

The email confirmation didn't have the "Other employees..." field at all.
image






One question:
- If I enter a current website and go back to that field, there is a second form field displayed. Is that intended? This happens without me trying to add a second website. 
image

**Comment by michelle-rago** [comment]
One question:
- If I enter a current website and go back to that field, there is a second form field displayed. Is that intended? This happens without me trying to add a second website. 
image

**Comment by dave-kennedy-ecs** [comment]
> One question:
> 
> * If I enter a current website and go back to that field, there is a second form field displayed. Is that intended? This happens without me trying to add a second website.
>   image

This is how it was initially developed.  I think we should bring in line with the other forms with repeatable fields, but it is a good bit of work to do.  And would need its own work item if we undertake it.

File: src/registrar/templates/application_review.html (hunk 2 above)
Comment (dave-kennedy-ecs) [reply src/registrar/templates/application_review.html]: I agree.  We considered this.  But left this edge case outside of summary_item for now.  Might be worth revisiting this decision at a later date.

**Comment by michelle-rago** [comment]
Dave - we're getting close. Couple things:

- The email shows "Yes" for election office. The post-submission review page shows "True." Can we change the review page to "Yes." 
- The email shows "State or territory" for the type of org. The review page shows the long answer. Can we show the short "State or territory" version on the review page? 
- The pre-submission review page also shows the long state/territory answer. Can we use the short one there too? (Third screenshot)

**Email screenshot**
image

**Post-submission review page screenshot**
image

**Pre-submission review page screenshot**
image

**Comment by dave-kennedy-ecs** [comment]
> Dave - we're getting close. Couple things:
> 
> * The email shows "Yes" for election office. The post-submission review page shows "True." Can we change the review page to "Yes."
> * The email shows "State or territory" for the type of org. The review page shows the long answer. Can we show the short "State or territory" version on the review page?
> * The pre-submission review page also shows the long state/territory answer. Can we use the short one there too? (Third screenshot)
> 
> **Email screenshot** image
> 
> **Post-submission review page screenshot** image
> 
> **Pre-submission review page screenshot** image

@michelle-rago great catches.  All three bullet points have been changed.

**Review by michelle-rago — APPROVED** [review]
Yay!

**Review by zandercymatics — APPROVED** [review]
LGTM, though see question

File: src/registrar/tests/test_views.py [hunk 3]
```diff
@@ -2360,18 +2360,6 @@ def test_long_org_name_in_application(self):
...
-        Make sure the long name is displaying in the application summary
-        page (manage your application)
-        """
-        completed_application(status=DomainApplication.ApplicationStatus.SUBMITTED, user=self.user)
-        home_page = self.app.get("/")
-        self.assertContains(home_page, "city.gov")
-        # click the "Edit" link
-        detail_page = home_page.click("Manage", index=0)
-        self.assertContains(detail_page, "Federal: an agency of the U.S. government")
-
```
[the remark this answers is not in the document]
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_views.py:2374]: correct

**Pull request merged**